src/components/NotificationList.tsx

```diff
@@ -0,0 +1,102 @@
+import { View, Text } from "@/src/components/Themed";
+import colors from "../constants/colors";
+import { useEffect, useState } from "react";
+import { getMockNotifications, NotificationData } from "../lib/mockData";
+import { NotifProfilePicture } from "./ProfilePicture";
+import Divider from "./Divider";
+import { HabitReminder, HabitInvite, FriendRequest } from "./Notifications";
+
+// fetch all invites a user has received
+export default function NotificationList() {
+  const [notifications, setNotifications] = useState<NotificationData[]>([]);
+
+  useEffect(() => {
+    const fetchNotifications = async () => {
+      // we would fetch the list from the db and set it here
+      const notifs = await getMockNotifications();
+      setNotifications(notifs);
+    };
+    fetchNotifications();
+  }, []);
+
+  const deleteNotification = (id: number) => {
+    console.log("Notification deleted");
+    // delete in db
+    setNotifications((prevNotifs) =>
+      prevNotifs.filter((notif) => notif.id !== id),
+    );
+  };
+  const confirmNotification = (id: number) => {
+    console.log("Notification confirmed");
+    // confirm in db
+    setNotifications((prevNotifs) =>
+      prevNotifs.filter((invite) => invite.id !== id),
+    );
+  };
+
+  if (notifications.length === 0) {
+    return (
+      <View className="">
+        <Text className="mt-8 text-center text-base font-medium text-stone-400">
+          You’re all caught up 🎉
+        </Text>
+
+      </View>
+    );
+  }
+  return (
+    <View>
+      <Divider/>
+      <View className="flex flex-col">
+        {notifications.map((notif, index) => (
+          <View>
+            <View className="px-4">
+              {(notif.type === "friendRequest") &&
+                <FriendRequest
+                key={index}
```

Comment:
The View on line 52 needs the key prop. Also you shouldn't use the index as a key (maybe add an id to the notification object and use that).